exec/cnex/support.c

```diff
@@ -0,0 +1,222 @@
+{
+    // Paths are arlready initialized, so just return.
+    if (neon_paths) {
+        return;
+    }
+
+    // Push neon executable path first.
+    if (strlen(source_path)) {
+        path_pushPath(source_path);
+    }
+    // Then push the cwd path.
+    path_pushPath("./");
+
+    // Get paths from %NEONPATH% environment variable, if there is one.
+    char *ne = getenv("NEONPATH");
+    if (ne != NULL) {
+        char *next = strtok(ne, PATHSEP);
+        while (next) {
+            path_pushPath(next);
+            next = strtok(NULL, PATHSEP);
+        }
+    }
+
+    // Next, check for .neonpath, and process any paths that might exist in it.
+    FILE *fp = fopen(".neonpath", "r");
+    if (fp) {
+        for (;;) {
+            char path[1024];
+            if (fgets(path, sizeof(path), fp) == NULL) {
+                break;
+            }
+            // ToDo: Trim possible leading and trailing spaces (whitespace) off loaded paths.
+            size_t len = strlen(path);
+            if (path[len-1] == '\n') {
+                path[len-1] = '\0';
+            }
+            // Don't bother pushing empty paths.
+            if (len > 0) {
+                path_pushPath(path);
+            }
+        }
+        fclose(fp);
+    }
+}
+ 
+void path_readModule(TModule *m)
+{
+    char modulePath[1024];
+
+    FILE *fp = path_findModule(m->name, modulePath);
+    if (fp == NULL) {
+        path_dumpPaths();
+        fatal_error("Could not find Neon module \"%s\"", m->name);
+    }
+    fseek(fp, 0, SEEK_END);
+    long nSize = ftell(fp);
+    fseek(fp, 0, SEEK_SET);
+
+    m->code = malloc(nSize);
+    if (m->code == NULL) {
+        fatal_error("Could not allocate %d bytes for bytecode from %s module.", nSize, m->name);
+    }
+
+    m->codelen = (unsigned int)fread(m->code, 1, nSize, fp);
+    fclose(fp);
+}
+
+FILE *path_findModule(const char *name, char *modulePath)
+{
+    TPath *next = neon_paths;
+    FILE *r = NULL;
+
+    char filename[1024];
+    while (next != NULL) {
+        // ToDo: Locate .neon*, files, and return full paths / handles to those as well.
+        sprintf(filename, "%s/%s.neonx", next->path, name);
+        r = fopen(filename, "rb");
+        if (r) {
+            if (modulePath) {
+                strcpy(modulePath, filename);
+            }
+            return r;
+        }
+        next = next->next;
+    }
+    return NULL;
+}
+
+void path_dumpPaths()
+{
+    if (0) {
+        TPath *next = neon_paths;
+
+        assert(next != NULL);
+
+        printf("cnex module search paths:\n-------------------------\n");
+        int count = 1;
+        while (next != NULL) {
+            printf("%d: %s\n", count++, next->path);
+            next = next->next;
+        }
+    }
+}
+
+char *path_getFileNameOnly(char *path)
+{
+    char *p = path;
+    char *s = p;
+
+    while (*p) {
+        if (*p == '\\' || *p == '/') {
+            s = p+1;
+        }
+        p++;
+    }
+    if (s[0] == '\\' || s[0] == '/') {
```

Comment:
Under what conditions could this ever be true? The above loop should skip past all slashes.